diambra/arena/engine/interface.py

```diff
@@ -35,67 +35,106 @@ def __init__(self, env_address, grpc_timeout=60):
     # Transforming env settings dict to pb request
     def env_settings_to_pb_request(self, env_settings):
 
-        characters = {
-            "p1": [env_settings["characters"][0][0], env_settings["characters"][0][1], env_settings["characters"][0][2]],
-            "p2": [env_settings["characters"][1][0], env_settings["characters"][1][1], env_settings["characters"][1][2]]
-        }
-        outfits = {
-            "p1": env_settings["char_outfits"][0],
-            "p2": env_settings["char_outfits"][1]
-        }
         frame_shape = {
-            "h": env_settings["frame_shape"][0],
-            "w": env_settings["frame_shape"][1],
-            "c": env_settings["frame_shape"][2]
-        }
-        action_spaces = {
-            "p1": model.ACTION_SPACE_DISCRETE if env_settings["action_space"][0] == "discrete" else model.ACTION_SPACE_MULTI_DISCRETE,
-            "p2": model.ACTION_SPACE_DISCRETE if env_settings["action_space"][1] == "discrete" else model.ACTION_SPACE_MULTI_DISCRETE,
-        }
-        attack_buttons_combinations = {
-            "p1": env_settings["attack_but_combination"][0],
-            "p2": env_settings["attack_but_combination"][1]
-        }
-        super_arts = {
-            "p1": env_settings["super_art"][0],
-            "p2": env_settings["super_art"][1]
-        }
-        fighting_styles = {
-            "p1": env_settings["fighting_style"][0],
-            "p2": env_settings["fighting_style"][1]
+            "h": env_settings.frame_shape[0],
+            "w": env_settings.frame_shape[1],
+            "c": env_settings.frame_shape[2]
         }
-        ultimate_styles = {
-            "p1": {
-                "dash": env_settings["ultimate_style"][0][0],
-                "evade": env_settings["ultimate_style"][0][1],
-                "bar": env_settings["ultimate_style"][0][2]
-            },
-            "p2": {
-                "dash": env_settings["ultimate_style"][1][0],
-                "evade": env_settings["ultimate_style"][1][1],
-                "bar": env_settings["ultimate_style"][1][2]
+
+        if env_settings.player == "P1P2":
+            characters = {
+                "p1": [env_settings.characters[0][0], env_settings.characters[0][1], env_settings.characters[0][2]],
+                "p2": [env_settings.characters[1][0], env_settings.characters[1][1], env_settings.characters[1][2]]
+            }
+            outfits = {
+                "p1": env_settings.char_outfits[0],
+                "p2": env_settings.char_outfits[1]
+            }
+            action_spaces = {
+                "p1": model.ACTION_SPACE_DISCRETE if env_settings.action_space[0] == "discrete" else model.ACTION_SPACE_MULTI_DISCRETE,
```

Comment:
Oh looks liek you even have types for this, so let's use these?

Reply:
As in the comment above: noted, this will be included in the rework that will make use of protobuffer class directly